.github/actions/build-docker/action.yml

```diff
@@ -1,114 +1,183 @@
+    cache_from:
+      required: false
+      description: Cache to pull before building the image
 
 outputs:
+  artifact_name:
+    description: the artifact where the image was uploaded toJson
+    value: ${{ steps.input_meta.outputs.artifact_name }}
   digest:
-    description: "The Docker image digest"
-    value: ${{ steps.digest.outputs.digest }}
-  version:
-    description: "The Docker version for the image"
-    value: ${{ steps.meta.outputs.version }}
-  tags:
-    description: "The Docker tags for the image"
-    value: ${{ steps.meta.outputs.tags }}
+    description: The image digest from the index.json
+    value: ${{ steps.build.outputs.digest }}
 
 runs:
-  using: "composite"
+  using: 'composite'
   steps:
-    # Setup docker to build for multiple architectures
-    - name: Set up QEMU
-      uses: docker/setup-qemu-action@v3
-    - name: Set up Docker Buildx
-      uses: docker/setup-buildx-action@v1
-      with:
-        version: latest
-        buildkitd-flags: --debug
+    - id: input_meta
+      shell: bash
+      run: |
+        artifact_name="${{ inputs.artifact_name }}"
+        registry="${{ inputs.registry }}"
+        image="${{ inputs.image }}"
+        version="${{ inputs.version }}"
+        target="${{ inputs.target }}"
+        username="${{ inputs.username }}"
+        password="${{ inputs.password }}"
+        push="${{ inputs.push }}"
+        cache_to="${{ inputs.cache_to }}"
+        cache_from="${{ inputs.cache_from }}"
 
-    # Login to a registry to push the image
-    - name: Login to Container Registry
-      # Only login if we are pushing the image
-      if: ${{ inputs.push == 'true' }}
-      uses: docker/login-action@v3
-      with:
-        username: ${{ inputs.username }}
-        password: ${{ inputs.password }}
+        echo "registry=$registry" >> $GITHUB_OUTPUT
+        echo "image_file=image.tar" >> $GITHUB_OUTPUT
+        echo "version=$version" >> $GITHUB_OUTPUT
+        echo "target=$target" >> $GITHUB_OUTPUT
+        echo "cache_to=$cache_to" >> $GITHUB_OUTPUT
+        echo "cache_from=$cache_from" >> $GITHUB_OUTPUT
+
+        if [[ "$target" != 'development' && "$target" != 'production' ]]; then
+          echo "invalid target $target. Expected `production` or `development`"
+          exit 1
+        else
+          echo "target=$target" >> $GITHUB_OUTPUT
+        fi
 
+        base_image="$registry/$image"
+        echo "base_image=$base_image" >> $GITHUB_OUTPUT
 
-    - name: Suffix for local builds
-      id: suffix
-      shell: bash
-      # The suffix is unique to a branch and commit
-      run: |
-        if [[ "${{ inputs.push }}" == "false" ]]; then
-          echo "suffix=${{ github.ref }}-${{ github.sha }}" >> $GITHUB_OUTPUT
+        if [[ "$push" == 'true' ]]; then
+          if [[ "$username" != '' && "$password" != '' ]]; then
+            should_push=true
+          else
+            echo "`push` set to true but missing username: $username and/or password: $password"
```

Comment:
Cannot use backtick for strings, this is executed apparently

https://github.com/mozilla/addons-server/actions/runs/9758341983/job/26932695197?pr=22433

Reply:
Doesn't this have the potential to print out the actual password when the `username` is empty and the `password` is not? 

_maybe we don't want to echo `$password` at all._